### PR TITLE
Remove unused files before archiving build dir

### PR DESCRIPTION
**Pull request by pzread**
The archive step of `build_all` job uses up almost all RAM disk space. In addition to enlarging the RAM disk, we can also remove the unused files before the archive step.

This only takes 1s to run and saves about 10G disk space for the following archive step:
```
Filesystem      Size  Used Avail Use% Mounted on
tmpfs            50G   35G   16G  69% /runner-root
```

### CODE REVIEW

**Comment by pzread** [comment]
`Tar` has another option called `--remove-files `. It seems like `tar` will delete the processed files during the archiving. But I think this is overkill as when we unarchive the `build-dir.tar` it will still take the space of tar size + unarchived build dir. So even we successfully finish `build_all`, the following jobs might run out of space when unarchiving.

**Review by jpienaar — APPROVED** [review]
LG (esp if this resolves the running out of build space one)

**Comment by pzread** [comment]
https://github.com/iree-org/iree/pull/10725 the LLVM bump seems to work with this change. I'd wait for the review from @GMNGeoffrey

File: .github/workflows/ci.yml
Comment (GMNGeoffrey) [inline .github/workflows/ci.yml:127]: Can also just use `-delete` I think?

```suggestion
          find "${BUILD_DIR}" -type f -name "*.a" -o -type f -name "*.o" -delete
```

**Review by GMNGeoffrey — APPROVED** [review]
This seems fine. Can you link an issue about the disk space issue here with a TODO. We really shouldn't have to be worrying about disk space (sorry that I configured it too low). I would want to increase the disk size and then revert this.

**Pull request merged**
